pinot-core/src/main/java/org/apache/pinot/core/operator/BaseProjectOperator.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.operator;
+
+import java.util.Map;
+import org.apache.pinot.common.request.context.ExpressionContext;
+import org.apache.pinot.core.operator.blocks.ValueBlock;
+
+
+public abstract class BaseProjectOperator<T extends ValueBlock> extends BaseOperator<T> {
```

Comment:
do we need the generic here? seems like the usages are always ValueBlock instead of the generic extension of `T` right? any chance we want to use it differently going forward?

Reply:
We don't right now, but I do want to make it more specific since `ValueBlock` is not a concrete class